src/containerapp/azext_containerapp/custom.py

```diff
@@ -690,6 +705,24 @@ def create_containerapp(cmd,
         else:
             set_managed_identity(cmd, resource_group_name, containerapp_def, user_assigned=[registry_identity])
     try:
+        image = None if imageNotProvided else _reformat_image(source,repo,image)
```

Comment:
I'm not sure I understand the relationship between `imageNotProvided` and `image` if we set them to `True` and `HELLO_WORLD_IMAGE`, respectively, at the beginning of this function, but then revert `image` back to `None` if `imageNotProvided` is `True`. Would you mind elaborating on the purpose of this line?

Reply:
Changed this line to `image = None if self.get_argument_image().__eq__(HELLO_WORLD_IMAGE) else _reformat_image(self.get_argument_source(), self.get_argument_repo(), self.get_argument_image())`

Since image is defaulted to `mcr.microsoft.com/k8se/quickstart:latest` if it was not provided by the user, we need to reassign it to None so the final image pushed to the registry is in the format `<registry-server>/<containerapp-name>:tag`